lib/blacklight/xml_parser.rb

```diff
@@ -113,7 +113,8 @@ def self.get_master_parent(pre_data, parents_ids, parent_id)
   # non-metadata file.
   ##
   def self.iterate_files(zipfile)
-    files = zipfile.glob("*/**/**")
+    files = zipfile.entries.select(&:file?)
```

Comment:
Also, it might be worth mentioning that it could depend on how the zip file was created. Some tools might generate zip files where directories have separate entries (like @nick-benoit14 is seeing) while others might generate zip files where the directories are implied by the file paths (and so wouldn't have explicit directory entries).